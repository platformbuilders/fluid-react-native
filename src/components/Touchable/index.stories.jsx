import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { withKnobs, text, boolean } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import Touchable from './index';
import touchableMarkdown from './touchable.md';

storiesOf('Touchable', module)
  .addDecorator(withKnobs)
  .addParameters({
    readme: {
      content: touchableMarkdown,
    },
    viewport: {
      defaultViewport: '',
    },
  })
  .add('Simple example', () => (
    <Touchable
      disabled={boolean('disabled', false)}
      onPress={action('button-click')}
    >
      {text('Text', 'Touchable Text')}
    </Touchable>
  ));
