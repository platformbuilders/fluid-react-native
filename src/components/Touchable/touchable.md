
# Touchable ðŸš€

<!-- STORY -->

## Source Code
<!-- SOURCE -->

## Props

| Name  | Required | Type | Default Value | Description 
| ------------- | ------------- | ------------- |------------- |------------- |
| id | false | string | ' '  |   |
| accessibility | false | string | ' '  |   |
| accessibilityLabel | false | string | ' '  |   |
| testID | false | string | ' '  |   |
| disabled | false | boolean | false  |   |
| onPress | false | function | (param: any) => void  |   |

