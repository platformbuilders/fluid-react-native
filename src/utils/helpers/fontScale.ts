import { RFValue } from 'react-native-responsive-fontsize';

export const scaledFontSize = (
  size: number | string,
  baselineFactor?: 280,
): number => {
  const value = typeof size === 'string' ? parseInt(size, 10) : size;
  return RFValue(value, baselineFactor);
};
