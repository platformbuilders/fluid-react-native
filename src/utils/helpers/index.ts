export {
  currencyParser,
  parseToThousands,
  removeWhiteSpaces,
  toOnlyNumbers,
  getTheme,
  ifStyle,
  switchStyle,
} from '@platformbuilders/helpers';
export * from './generateHaptic';
export * from './getByTypographyVariant';
export * from './fontScale';
export * from './shadow';
