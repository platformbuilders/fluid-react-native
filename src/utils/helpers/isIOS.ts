import { Platform } from 'react-native';

const IOS_PLATFORM = 'ios';

export const isIOS = (): boolean => Platform.OS === IOS_PLATFORM;
