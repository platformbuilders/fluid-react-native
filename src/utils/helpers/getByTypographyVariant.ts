import { getTheme } from '@platformbuilders/helpers';
import { scaledFontSize } from './fontScale';

export const getFontSize = (props: any): number => {
  const fontSize = getTheme(`${props.variant}.fontSize`)(props) || 16;
  return scaledFontSize(fontSize);
};

export const getLineHeight = (props: any): number => {
  const lineHeight = getTheme(`${props.variant}.lineHeight`)(props) || 22;
  return scaledFontSize(lineHeight);
};
