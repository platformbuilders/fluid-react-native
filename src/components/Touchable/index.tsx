import React, { FC } from 'react';
import { TouchableOpacity } from 'react-native';
import { TouchableType } from '../../types';
import { generateHaptic } from '../../utils/helpers';

const CommonTouchable: FC<TouchableType> = ({
  onPress = (): void => {},
  haptic = 'impact',
  disabled = false,
  accessibility,
  accessibilityLabel,
  testID,
  id,
  ...rest
}) => (
  <TouchableOpacity
    {...rest}
    accessibilityLabel={accessibilityLabel || accessibility}
    testID={testID || id || accessibility}
    disabled={disabled}
    onPress={(e): void => {
      generateHaptic(haptic);
      onPress(e);
    }}
  />
);

export default CommonTouchable;
