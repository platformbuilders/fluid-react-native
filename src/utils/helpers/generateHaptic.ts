import Haptic, { HapticFeedbackType } from 'react-native-haptic';
import { isIOS } from './isIOS';

export const generateHaptic = (haptic: HapticFeedbackType = 'impact'): void => {
  if (isIOS()) {
    Haptic.generate(haptic);
  }
};
