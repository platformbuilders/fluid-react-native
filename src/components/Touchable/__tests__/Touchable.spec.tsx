import 'jest';
import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from 'react-native-testing-library';
import { ThemeProvider } from 'styled-components/native';
import Touchable from '..';
import { theme } from '../../../test/helpers';

describe('<Touchable />', () => {
  it('should render Touchable', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <Touchable id="testing" accessibility="" />
      </ThemeProvider>,
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should fire onPress from Touchable', () => {
    const onPressEvent = jest.fn();

    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <Touchable id="testing" accessibility="" onPress={onPressEvent} />
      </ThemeProvider>,
    );

    const component = getByTestId('testing');

    fireEvent.press(component);

    expect(onPressEvent).toHaveBeenCalled();
  });
});
