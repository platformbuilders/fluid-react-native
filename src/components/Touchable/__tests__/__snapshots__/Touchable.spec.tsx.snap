// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Touchable /> should render Touchable 1`] = `
<View
  accessibilityLabel=""
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="testing"
/>
`;
